modules/nanopb/nanopb.cmake

```diff
@@ -30,4 +32,12 @@ function(zephyr_nanopb_sources target)
 
   target_include_directories(${target} PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
   target_sources(${target} PRIVATE ${proto_srcs} ${proto_hdrs})
+
+  foreach(hdr ${proto_hdrs})
+    generate_unique_target_name_from_filename(${hdr} hdr_target_name)
+    add_custom_target(${hdr_target_name} DEPENDS ${hdr})
+
+    add_dependencies(nanopb_generated_headers ${hdr_target_name})
+  endforeach()
```

Comment:
why a for loop here ?
Isn't it enough to just have a single custom target which can then depend on the list of generated headers ?

Secondly, though I haven't explored fully the implementation of `nanopb_generate_cpp()`, then it does seem like there could be a missing knowledge on how headers are actually generated, as the initial look I did doesn't seem to have them as output or byproducts anywhere.

Reply:
I didn't know how else to create a target for a list of files...

You mean the `nanopb_generate_cpp` needs adjustments? I probably lack the cmake experience here to know what would be best.

Reply:
seems you did manage to have a single target in the updated version.

Reply:
Yes, I think this should be fine as the list of header files is used to generate a unique md5 hash now.